### PR TITLE
fix: PartialUpdateObject() throwing "data should not be IEnumerable" with JObject

### LINKED ISSUE #654: bug: saveObject throwing "data should not be an IEnumerable" with a single JObject
- Algolia Client Version: 6.4.0
- Language Version: .NET Standard 2?0

### Description

The `saveObject` method is throwing the following exception when trying the following (valid) snippet:

`data should not be an IEnumerable/List/Collection`

```csharp
var foo = new JObject();
foo.Add("objectID","10");
index.SaveObject(foo);
```

It's because `JObject` inherits from `IEnumerable`

### Steps To Reproduce

```csharp
var foo = new JObject();
foo.Add("objectID","10");
index.SaveObject(foo);
```

## Solution

Improve the [following check](https://github.com/algolia/algoliasearch-client-csharp/blob/9154becc34a3eb4d98cd6fe7ebbbfe90721618e2/src/Algolia.Search/Clients/SearchIndex.cs#L154) to allow JObject in saveObject

## Work-around

Wrapping the `JObject` in a `JArray` and using `SaveObjects` can be used as a work-around while waiting for the fix to be published.

```csharp
var foo = new JObject {{"objectID", "10"}};
var array = new JArray {foo};
index.SaveObjects(array);
```

### PR DESCRIPTION
**Pull request by TomKlotzPro**
| Q                 | A
| ----------------- | ----------
| Bug fix?          | yes
| New feature?      | yes/no   
| BC breaks?        | no     
| Related Issue     | Fix #654
| Need Doc update   | no

## Describe your change

This commit adds additional check in the PartialUpdateObject() method, to allow JObject as parameter.

### CODE REVIEW

**Comment by Ant-hem** [comment]
@TomKlotzPro Could you please add a UT?

File: src/Algolia.Search.Test/EndToEnd/Index/IndexingTest.cs [hunk 1]
```diff
@@ -282,6 +288,27 @@ public async Task MoveIndexTest()
             Assert.True(listIndices.Items.Exists(x => x.Name.Equals(indexDestName)));
             Assert.False(listIndices.Items.Exists(x => x.Name.Equals(_indexMoveName)));
         }
+
+        [Test]
+        [Parallelizable]
+        public async Task IndexOperationsAsyncWithJObjectTest()
```
Comment (Ant-hem) [inline src/Algolia.Search.Test/EndToEnd/Index/IndexingTest.cs:294]: Good test. Could you please also test the case when the `JObject` doesn't have an `objectID`?

**Review by Ant-hem — APPROVED** [review]
LGTM. Good work @TomKlotzPro.

File: src/Algolia.Search/Utils/AlgoliaHelper.cs [hunk 2]
```diff
@@ -101,6 +101,18 @@ internal static void EnsureObjectID<T>()
         /// <returns></returns>
         internal static string GetObjectID<T>(T data)
         {
+            Type itemType = typeof(T);
+
+            if (itemType == typeof(JObject))
+            {
+                JProperty objectID = JObject.FromObject(data).Property("objectID");
+
+                if (objectID != null && objectID.ToString() != "")
```
Comment (Ant-hem) [inline src/Algolia.Search/Utils/AlgoliaHelper.cs:110]: cosmetic: you could have used String.IsNullOrEmpty(objectID.ToString()). Feel free to do it in a second PR.
reference: https://docs.microsoft.com/en-us/dotnet/api/system.string.isnullorempty?view=netcore-3.1

**Comment by NoelOConnell** [comment]
Is there a timeline for when this PR will be released in the next NuGet version?

**Pull request merged**